go-controller/pkg/ovn/ha_master.go

```diff
@@ -3,47 +3,73 @@ package ovn
 import (
 	"context"
 	"os"
+	"strconv"
+	"strings"
 	"time"
 
 	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/config"
 	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/factory"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/util"
 	"github.com/sirupsen/logrus"
+	kapi "k8s.io/api/core/v1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
 	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
 	"k8s.io/client-go/tools/leaderelection"
 	"k8s.io/client-go/tools/leaderelection/resourcelock"
 )
 
 const (
+	invalidIPAddress = "0.0.0.1"
 	haLeaderLockName = "ovn-kubernetes-master"
+	ovnkubeDbEp      = "ovnkube-db"
+	haMasterLeader   = "k8s.ovn.org.ovnkube-master-leader"
 )
 
 // HAMasterController is the object holder for managing the HA master
 // cluster
 type HAMasterController struct {
-	kubeClient    kubernetes.Interface
-	ovnController *Controller
-	nodeName      string
-	isLeader      bool
-	leaderElector *leaderelection.LeaderElector
-	stopChan      chan struct{}
+	kubeClient      kubernetes.Interface
+	ovnController   *Controller
+	manageDBServers bool
+	nodeName        string
+	isLeader        bool
+	leaderElector   *leaderelection.LeaderElector
+	stopChan        chan struct{}
 }
 
 // NewHAMasterController creates a new HA Master controller
 func NewHAMasterController(kubeClient kubernetes.Interface, wf *factory.WatchFactory,
-	nodeName string, stopChan chan struct{}) *HAMasterController {
+	nodeName string, stopChan chan struct{}, manageDBServers bool) *HAMasterController {
 	ovnController := NewOvnController(kubeClient, wf)
 	return &HAMasterController{
-		kubeClient:    kubeClient,
-		ovnController: ovnController,
-		nodeName:      nodeName,
-		isLeader:      false,
-		leaderElector: nil,
-		stopChan:      stopChan,
+		kubeClient:      kubeClient,
+		ovnController:   ovnController,
+		nodeName:        nodeName,
+		manageDBServers: manageDBServers,
```

Comment:
set `manageDBServers:` to `config.MasterHA.ManageDBServers` directly at L51

Reply:
Done